_targets.R

```diff
@@ -1,45 +1,95 @@
 library(targets)
 source("1_fetch/src/get_nwis_data.R")
+source("2_process/src/bind_data.R")
 source("2_process/src/process_and_style.R")
 source("3_visualize/src/plot_timeseries.R")
 
 options(tidyverse.quiet = TRUE)
-tar_option_set(packages = c("tidyverse", "dataRetrieval")) # Loading tidyverse because we need dplyr, ggplot2, readr, stringr, and purrr
 
-p1_targets_list <- list(
-  tar_target(
-    site_data,
-    download_nwis_data(),
-  ),
-  tar_target(
-    site_info_csv,
-    nwis_site_info(fileout = "1_fetch/out/site_info.csv", site_data),
-    format = "file"
+tar_option_set(packages = c(
+  "tidyverse", 
+  "dataRetrieval",
+  "utils",
+  "futile.logger"
   )
 )
 
-p2_targets_list <- list(
+# site number list 
+site_list <- c("01427207", "01432160", "01436690", "01466500")
+
+# establish common start date, end date, and parameter code for queries
+startDate <- "2014-05-01" 
+endDate <- "2015-05-01"
+parameterCd <- '00010'
+
+# Create output directories if not included
+if(!dir.exists('1_fetch/out/'))
+  dir.create(path = '1_fetch/out/')
+if(!dir.exists('2_process/out/'))
+  dir.create(path = '2_process/out/')
+if(!dir.exists('3_visualize/out/'))
+  dir.create(path = '3_visualize/out/')
+
+
+
+# Download site data and compile files
+p1_targets_list <- list(
   tar_target(
-    site_data_clean, 
-    process_data(site_data)
+    site_data,
+      for (site_num in site_list) {
```

Comment:
I think the issue you are having is that it is looping within the target and the end of the loop isn't returning anything. Whenever you have a file or directory that you want to be a target, your function must `return()` the character string of the file or directory. So much like you have done `return(out_file)`, I think you would also need `return(out_dir)` here but outside of the loop. 

That being said, a loop is not exactly the way I would approach this. The problem with the loop is that if anything in `site_list` changes (new one added, one deleted, one changed, etc), you are redownloading for everything in the list even if the site remained unchanged. So, you almost need to write the code to treat each target as a single site, so that they are only re-downloaded if that one site is new. Because we have only 4 sites, I think it is OK to write and copy code for each target. Pipelines III will go over how to scale beyond 4 sites when that approach is not an feasible.

So, I think you could write this as:

```r
tar_target(
    site_1_data,
    download_nwis_data(
        site_list[1],
        "1_fetch/out/", # out directory
        startDate,
        endDate,
        parameterCd
    ),
    format="file"
),
tar_target(
    site_2_data,
    download_nwis_data(
        site_list[2],
        "1_fetch/out/", # out directory
        startDate,
        endDate,
        parameterCd
    ),
    format="file"
)
```

and then have `download_nwis_data()` return the out filepath. This would mean that the `compiled_nwis_data` target would need to change to handle these individual file targets, too.

Reply:
Ok, this makes sense! Thank you!